en/03_Drawing_a_triangle/04_Swap_chain_recreation.md

```diff
@@ -53,8 +45,6 @@ void recreateSwapChain() {
 
     createSwapChain();
     createImageViews();
-    createRenderPass();
```

Comment:
I'm not actually sure we want to get rid of renderpass re-creation. 
The reason is that there is a non-zero chance of the Swapchain Image Format changing. I can imagine a weird setup where the window gets moved between a HDR and SDR monitor, and the window system requests a format change. Or something equally ridiculous. On windows and x11, I think these are unlikely but with wayland and more exotic window managers I don't have such confidence. 

I may be raising an issue that simply doesn't exist out in the wild, but its one thing that I would love some reassurance that "it doesn't happen in practice" so that this code isn't very subtly wrong.

Reply:
That's a good point, but tbh in all my years doing Vulkan I have never encountered such a situation, so I'd prefer to not have to recreate the render pass here. I don't do this in my samples either.

Adding notes on artificial cases may also irritate readers.

To sum it up: I'm not sure on this and open for discussion :)

Reply:
As per today's call: I'll reimplement this to make sure the sample is 100% correct, even in special cases.

Reply:
Apologies for making more work for you, but I think removing the recreation of renderpasses is okay as long as it is noted that we use swapchain image formats which are pretty much always supported. If a different format is used (eg, HDR) care is needed to ensure that the formats being used are always correct, no trying to render SDR to an HDR swapchain and vice versa.

Reply:
No problem at all :) I'll add a note on when render pass re-creation may be necessary, so for those few occasions where it can happen people at least have a pointer as to why it could happen.

Reply:
I have added a note on renderpass recreation:

> Note that we don't recreate the renderpass for simplicity. In theory it can be possible for the swap chain image format to change during an applications' life-time, e.g. when moving a window from an standard range to an high dynamic range monitor. This may require the application to recreate the renderpass to make sure the change between dynamic ranges is properly reflected.

Hope that fits into the tutorial and is easy (enough) to understand. Let me know if you're fine with that :)